src/drivers/uavcan/actuators/servo.cpp

```diff
@@ -60,3 +88,64 @@ UavcanServoController::update_outputs(bool stop_motors, uint16_t outputs[MAX_ACT
 
 	_uavcan_pub_array_cmd.broadcast(msg);
 }
+
+void
+UavcanServoController::servo_temperature_sub_cb(const
+		uavcan::ReceivedDataStructure<uavcan::equipment::device::Temperature>
+		&msg)
+{
+	if (msg.device_id < servo_temperature_s::CONNECTED_SERVO_MAX) {
+		auto &ref = _servo_temperature.servo[msg.device_id];
+
+		ref.timestamp   = hrt_absolute_time();
+		ref.node_id     = msg.getSrcNodeID().get();
+		ref.actuator_id = msg.device_id;
+		ref.temperature = msg.temperature;
+		ref.error_flags = msg.error_flags;
+
+		_servo_temperature.counter += 1;
+		_servo_temperature.timestamp = hrt_absolute_time();
+		_servo_temperature_pub.publish(_servo_temperature);
```

Comment:
this callback will handle the `uavcan.equipment.device.Temperature` message from any node that publishes it, but here you're publishing it as a ServoTemperature.msg. Renaming would make it more clear:

ServoTemperature.msg --> DronecanDeviceTemperature.msg